attributes/network/cddl/9_account.cddl

```diff
@@ -0,0 +1,120 @@
+; tag::types[]
+account-role = "owner"
+
+; The attribute argument contains a list of all account features supported by
+; the server.
+account@attribute-arg = [ * uint ]
+; end::types[]
+
+; tag::create[]
+; Create a new account.
+account.create@param = {
+    ; A name or description of the account. This does not have to be unique.
+    ; The server is welcome to limit the name of a wallet in any ways it sees
+    ; fit.
+    ? 0 => tstr,
+
+    ; An optional initial list of roles. The sender of this message will be
+    ; included in the list of owners by default, even if the sender is listed
+    ; in this list.
+    ; The server MUST return an error if any role is not supported by any
+    ; feature included in the account.
+    ? 1 => {
+        * non-anonymous-identity => [ * tstr ],
+    },
+
+    ; Required list of features and their arguments.
+    ; The server MUST return an error if no features are selected; an account
+    ; without features doesn't do anything and is an error.
+    2 => [ 1* attribute ],
+
+    ; Any other information pertinent to this account.
+    * tstr => any,
+}
+account.create@return = {
+    ; The account identity. Use this to contact the account directly.
+    0 => identity,
+}
+; end::create[]
+
+; tag::setDescription[]
+; Sets the description of an account.
+account.setDescription@param = {
+    ; The identity of the account.
+    0 => non-anonymous-identity,
+
+    ; The new name or description.
+    1 => tstr,
+}
+
+account.account.setDescription@return = ()
+; end::setDescription[]
+
+; tag::listRoles[]
+; List all the roles available for an account, based on its features.
+account.listRoles@param = ()
+
+account.listRoles@return = {
+    ; The list of roles available for identities.
+    0 => [ * account-role ],
```

Comment:
Is there a way in CDDL to make it clearer that `account-role` can be `owner`, or any other roles defined in other spec?

Reply:
Normally if you're interested in the spec of a feature it would be clear from that feature's CDDL. But I'll add a comment.